### PR TITLE
cp2k: add new version and tweak settings

### PR DESCRIPTION
**Pull request by glennpj**
The @2022: versions use different settings than before when using CUDA.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
Waiting for @dev-zero to :heavy_check_mark: before merging

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -577,6 +587,10 @@ def edit(self, spec, prefix):
                     "-DOFFLOAD_TARGET=cuda",
                 ]
                 libs += ["-lcublas"]
+
+                if spec.satisfies("+cuda_fft"):
+                    cppflags += ["-D__PW_CUDA"]
+                    libs += ["-lcufft"]
```
Comment (dev-zero) [inline var/spack/repos/builtin/packages/cp2k/package.py:593]: you disabled the availability of the `cuda_fft` variant above I think

Comment (alalazo) [reply var/spack/repos/builtin/packages/cp2k/package.py:593]: From the perspective of Spack directives, this condition branch is entered only for version `9.*` when `+cuda_fft`. Not sure if this makes sense for `cp2k` though.

Comment (dev-zero) [reply var/spack/repos/builtin/packages/cp2k/package.py:593]: @alalazo exactly what I meant. In this PR there is also a `when="@:9",  # req in CP2K v2022+` added for the `cuda_fft` variant.

Comment (glennpj) [reply var/spack/repos/builtin/packages/cp2k/package.py:593]: I do not think `cuda_fft` is disabled above as it is enabled by default in `@2022:`. That is why I set the condition on the variant. However, looking at the cp2k docs again, I see that it _can_ be disabled in the 2022 version. If I am understanding correctly, versions up to 9 would use `D__PW_CUDA` to turn it on, and versions 2022 and above would use `D__NO_OFFLOAD_PW` to turn it off. So, I should redo the logic to reflect that. Does that sound correct @dev-zero?

Comment (glennpj) [reply var/spack/repos/builtin/packages/cp2k/package.py:593]: I added a commit to enable the `cuda_fft` variant for `@2022:` and adjusted the `if...` logic.

Comment (dev-zero) [reply var/spack/repos/builtin/packages/cp2k/package.py:593]: I think that looks right, yes.
Thanks a lot!

**Review by dev-zero — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Pull request merged**